### PR TITLE
Kubernetes docs for the 2.6 release

### PR DESCRIPTION
**Pull request by marccampbell**

### CODE REVIEW

File: docs/getting-started/replicated-private-registry.md [hunk 1]
```diff
@@ -53,3 +53,19 @@ Pushing tag for rev [8e471642d573] on {https://registry.replicated.com/v1/reposi
 
 For additional information on building, tagging and pushing docker images, please refer to the
 [Docker CLI Documentation](https://docs.docker.com/reference/commandline/cli/).
+
+## Deploying to Kubernetes
+
+When deploying an applucation to a [Kubernetes](/packaging-an-application/kubernetes) cluster, Replicated will automatically deploy a secret named `replicatedregistrykey`. This secret can be used as an `imagePullSecret` to gain read-only access to the images from the on-prem environment.
```
Comment (emosbaugh) [inline docs/getting-started/replicated-private-registry.md]: application

File: docs/kb/developer-resources/zero-downtime-backup.md [hunk 2]
```diff
@@ -43,30 +36,20 @@ components:
...
- command:
- - mv
- - /data/dump.rdb
- - /backup/dump.rdb
- run_type: exec
- component: DB
- image:
-   image_name: redis
-   version: latest
+   command: [mv, /data/dump.rdb, /backup/dump.rdb]
```
Comment (emosbaugh) [inline docs/kb/developer-resources/zero-downtime-backup.md]: indentation

File: docs/packaging-an-application/admin-commands.md [hunk 3]
```diff
@@ -73,8 +85,15 @@ Specify `exec` to execute the command in the currently running container. This i
 the only option.
 
 ### component
-Identifies the component under which the container image is defined.
+Required only when using the Replicated scheudler. Unavailable in other schedulers. This identifies the component under which the container image is defined.
```
Comment (emosbaugh) [inline docs/packaging-an-application/admin-commands.md]: scheduler

File: docs/packaging-an-application/admin-commands.md [hunk 4]
```diff
@@ -73,8 +85,15 @@ Specify `exec` to execute the command in the currently running container. This i
...
 ### image
-Identifies the image whose container will be used to run the command. This will not create a new
-instance of the container; the command will run inside the existing container.
+Required only when using the Replicated scheduler. Unavailable in other schedulers. This identifies the image whose container will be used to run the command. This will not create a new instance of the container; the command will run inside the existing container.
+
+### selectors
+Required only when using the Kubernetes scheduler. Unavailable in other schedulers. This is a standard Kubernetes selector to identify the component that the admin command should be run on.
+
+### container
+Available only when using the Kubernetes scheduler. Unavailable in other schedulers. This is to specify the pod to run the admin command on in in the resource that matches the selector. If not supplied, the admin command will run on the first pod in the resource.
```
Comment (emosbaugh) [inline docs/packaging-an-application/admin-commands.md]: container is available in both schedulers. required in the replicated scheduler

File: docs/packaging-an-application/custom-metrics.md [hunk 5]
```diff
@@ -155,9 +148,7 @@ The CSS can be added in Vendor Web dashboard in [App Settings](https://vendor.re
 
 ## Integration API
 
-StatsD host can be discovered via Replicated Console Settings. The following Go function will
-return the current end point. Note that this value can change, and applications should periodically
-query Integration API to obtain the current end point.
+StatsD host can be discovered via Replicated Console Settings. The following Go function will return the current end point. Note that this value can change, and applications should periodically query Integration API to obtain the current end point.
```
Comment (emosbaugh) [inline docs/packaging-an-application/custom-metrics.md]: endpoint

File: docs/packaging-an-application/kubernetes.md [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+  run_type: exec
+  selectors:
+    app: redis
+    tier: backend
+    role: master
+  container: master # optional, will choose first in pod
+```
+
+### Dashboard Metrics
+When running Replicated in Kubernetes, the standard statsd endpoint is still running. The only difference here is that the standard CPU and Memory usage graphs will be be available. You can use the [custom metrics](/packaging-an-application/custom-metrics) feature to define you own application-specific metrics to show on the admin console dashboard.
```
Comment (emosbaugh) [inline docs/packaging-an-application/kubernetes.md]: will not be available

Comment (ghost) [inline docs/packaging-an-application/kubernetes.md]: usage graphs will _not_ be available

File: docs/packaging-an-application/kubernetes.md [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+
+### Ready State/Health Checks
+Replicated will consider the application running when all of the Kubernetes resources are running. Different resources types have various methods to determine when they are started.
+
+| **Resource Type** | **Replicated Considers Running When...** |
+|-----|----|
+| Deployments | Same as [Kubernetes rollout status] (https://kubernetes.io/docs/user-guide/kubectl/kubectl_rollout_status/) |
+| Replication Controller | Same as [Kubernetes rollout status](https://kubernetes.io/docs/user-guide/kubectl/kubectl_rollout_status/) |
```
Comment (emosbaugh) [inline docs/packaging-an-application/kubernetes.md:61]: im not sure whether or not rollout status applies to rcs but good enough for now

File: docs/packaging-an-application/kubernetes.md [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+When running Replicated in Kubernetes, the standard statsd endpoint is still running. The only difference here is that the standard CPU and Memory usage graphs will be be available. You can use the [custom metrics](/packaging-an-application/custom-metrics) feature to define you own application-specific metrics to show on the admin console dashboard.
+
+### Ready State/Health Checks
+Replicated will consider the application running when all of the Kubernetes resources are running. Different resources types have various methods to determine when they are started.
+
+| **Resource Type** | **Replicated Considers Running When...** |
+|-----|----|
+| Deployments | Same as [Kubernetes rollout status] (https://kubernetes.io/docs/user-guide/kubectl/kubectl_rollout_status/) |
+| Replication Controller | Same as [Kubernetes rollout status](https://kubernetes.io/docs/user-guide/kubectl/kubectl_rollout_status/) | 
+| Persistent Volume Claim | When the volume in bound |
```
Comment (emosbaugh) [inline docs/packaging-an-application/kubernetes.md]: When the claim is bound

File: docs/packaging-an-application/kubernetes.md [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+
+### Ready State/Health Checks
+Replicated will consider the application running when all of the Kubernetes resources are running. Different resources types have various methods to determine when they are started.
+
+| **Resource Type** | **Replicated Considers Running When...** |
+|-----|----|
+| Deployments | Same as [Kubernetes rollout status] (https://kubernetes.io/docs/user-guide/kubectl/kubectl_rollout_status/) |
+| Replication Controller | Same as [Kubernetes rollout status](https://kubernetes.io/docs/user-guide/kubectl/kubectl_rollout_status/) | 
+| Persistent Volume Claim | When the volume in bound | 
+| Service | When type is set to LoadBalancer, it's running when the IP address is assigned. |
```
Comment (emosbaugh) [inline docs/packaging-an-application/kubernetes.md]: you can add ingress too. its the same. when a loadbalancer ip is assigned

File: docs/packaging-an-application/kubernetes.md [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+```bash
+SCHEDULER_ENGINE=kubernetes
+K8S_OVERRIDE_CLUSTER_CONFIG=true
+K8S_HOST=https://10.10.10.10
+K8S_TLS_INSECURE=true
+K8S_USERNAME=admin
+K8S_PASSWORD=password
+K8S_CLIENT_CERT_DATA="-----BEGIN CERTIFICATE-----…"
+K8S_CLIENT_KEY_DATA="-----BEGIN RSA PRIVATE KEY-----…"
+K8S_CLUSTER_CA_CERT_DATA="-----BEGIN CERTIFICATE-----..."
```
Comment (emosbaugh) [inline docs/packaging-an-application/kubernetes.md]: …

File: docs/distributing-an-application/installing-on-kubernetes.md [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+
+```shell
+kubectl apply -f https://get.replicated.com/kubernetes.yml
+```
+
+## Prerequisites
+The Kubernetes cluster must already be provisioned.
+
+### Volumes
+It's recommended to create a persistent volume on the cluster named `replicated-pv` with a minimum size of 10GB. If you are running Replicated on GKE, you can do this with the following command:
```
Comment (ghost) [inline docs/distributing-an-application/installing-on-kubernetes.md]: Needs another PV, `replicated-statsd-pv`

GCloud command:

`$ gcloud compute disks create --size=10GB --zone=<zone> replicated-statsd-pv`

**Pull request merged**